File: winanalytics/src/main/java/com/winfooz/MapUtils.java
package com.winfooz;

import java.util.HashMap;
import java.util.Map;
import org.json.JSONObject;

/**
 * Project: WinAnalytics2 Created: November 16, 2018
 *
 * @author Mohamed Hamdan
 */
public class MapUtils {

    public static JSONObject toJsonObject(HashMap<String, Object> map) {
        JSONObject object = new JSONObject();
        for (Map.Entry<String, Object> entry : map.entrySet()) {
            try {
                object.put(entry.getKey(), entry.getValue());
            } catch (Exception ignored) {
            }
        }
        return object;
    }
}

File: winanalytics/src/main/java/com/winfooz/MapUtils.kt
package com.winfooz

import android.os.Bundle
import org.json.JSONObject
import java.util.HashMap

/**
 * Project: WinAnalytics2 Created: November 16, 2018
 *
 * @author Mohamed Hamdan
 */
fun <V> HashMap<String, V?>.toJsonObject(): JSONObject {
    val json = JSONObject()
    entries.forEach {
        json.put(it.key, it.value)
    }
    return json
}

fun <V> HashMap<String, V?>.toBundle(): Bundle {
    val bundle = Bundle()
    entries.forEach {
        bundle.putString(it.key, it.value.toString())
    }
    return bundle
}